lxd/network/openvswitch/ovs.go

```diff
@@ -143,3 +143,23 @@ func (o *OVS) InterfaceAssociateOVNSwitchPort(interfaceName string, ovnSwitchPor
 
 	return nil
 }
+
+// ChassisID returns the local chassis ID.
+func (o *OVS) ChassisID() (string, error) {
+	// ovs-vsctl's get command doesn't support its --format flag, so we always get the output quoted.
+	// However ovs-vsctl's find and list commands don't support retrieving a single column's map field.
+	// And ovs-vsctl's JSON output is unfriendly towards statically typed languages as it mixes data types
+	// in a slice. So stick with "get" command and use Go's strconv.Unquote to return the actual values.
```

Comment:
For that part, note that you could use a `[]interface{}` and then a switch statement for the type of the entry.
Still quite a bit of handling but better than dealing with raw JSON :)

Reply:
Yeah I considered that and would have certainly gone that way had I been needing to access more than one column. In this case its returned as a simple quoted string rather than json so this approach seemed simpler, as frustratingly even requesting a single column returns a whole set of sub structures which makes parsing it rather prolonged.